### PR TITLE
cleanup: send query params to arsenal auth method

### PR DESCRIPTION
**Pull request by rahulreddy**

### CODE REVIEW

**Comment by rahulreddy** [comment]
End to end http://ci.ironmann.io/gh/scality/Integration/9232

File: lib/utapi/utilities.js [hunk 1]
```diff
@@ -49,7 +49,11 @@ function _listBucketMetrics(host,
             }
         });
     });
-    auth.client.generateV4Headers(request, '', accessKey, secretKey, 's3');
+    // TODO: cleanup with refactor of generateV4Headers
+    request.path = '/buckets';
+    auth.client.generateV4Headers(request, { Action: 'ListMetrics' },
+        accessKey, secretKey, 's3');
+    request.path = '/buckets?Action=ListMetrics';
```
Comment (electrachong) [inline lib/utapi/utilities.js]: sorry it is not obvious to me : why do you change the request path after setting it before generating the headers?

Comment (rahulreddy) [inline lib/utapi/utilities.js]: So, the problem was that `auth.client.generateV4Headers` expects `request.path` to not have any query params in it, but we cannot make a HTTP request to Utapi without those query params.
We would be refactoring the Arsenal method soon, so for now, we remove the query params before generating the headers and put them back for making the request.

Comment (electrachong) [inline lib/utapi/utilities.js]: ok

**Comment by DavidPineauScality** [comment]
Build finally succeeded (after a few flaky failures and retries) at https://ci.ironmann.io/gh/scality/Integration/9251#build-parameters
Merging.

**Pull request merged**